run_trainer.sh

```diff
@@ -0,0 +1,150 @@
+CONDA_ENV_NAME=$1
+TESTNAME=$2
+TEST_SETUP=$3
+
+#RESUME_JOB_ID for checkpoint restore
+if [[ -z "${RESUME_JOB_ID}" ]]; then
+  JOB_ID=${SLURM_JOB_ID} 
+else
+  JOB_ID="${RESUME_JOB_ID}"
+fi
+
+#export NCCL_DEBUG="info" #For debugging NCCL and EFA 
+
+# # Editable paths
+
+# Source CONDA
+CONDA_HOME="/home/thiamha/miniconda3/" #TODO : Make this $HOME once everyone has setup miniconda
+source ${CONDA_HOME}/bin/activate ${CONDA_ENV_NAME}
+
+which python
+# # VENV
+# PY_VENV_PATH="/shared/apoorvgu/jax-21/bin/activate"
+# source ${PY_VENV_PATH}
+
+# NEURON_DUMP_PATH=${PWD}/neuron_dump
+# HLO_DUMP_PATH=${PWD}/hlo_dump
+
+# # Install runtime and collectives library. This is only needed in internal dev cluster
+# # Remove this before release
+# source ./bigcluster_setup.sh
+
+# # Neuron compiler flags
+# export NEURON_CC_FLAGS="--framework=XLA"
+# export NEURON_CC_FLAGS="${NEURON_CC_FLAGS} --model-type transformer"
+# export NEURON_CC_FLAGS="${NEURON_CC_FLAGS} --no-internal-hlo-remat"
+# export NEURON_CC_FLAGS="${NEURON_CC_FLAGS} --distribution-strategy=llm-training"
+# # export NEURON_CC_FLAGS="${NEURON_CC_FLAGS} --internal-hlo2tensorizer-options='--verify-hlo --num-concat-graphs=8'" # Set indside fuji.py with gradient_accumulation size
+# export NEURON_CC_FLAGS="${NEURON_CC_FLAGS} --enable-mixed-precision-accumulation"
+# export NEURON_CC_FLAGS="${NEURON_CC_FLAGS} -O1"
+# export NEURON_CC_FLAGS="${NEURON_CC_FLAGS} --dump=${NEURON_DUMP_PATH}"
+
+# # Neuron PJRT flags
+# export NEURON_WHILE_LOOP_UNROLL=1
+# export NEURON_RUN_TRIVIAL_COMPUTATION_ON_CPU=1
+
+# # Neuron runtime flags
+# export NEURON_RT_ASYNC_EXEC_MAX_INFLIGHT_REQUESTS=1
+
+# Neuron env vars for distributed training based on SLURM
+nodes=$(scontrol show hostnames "$SLURM_JOB_NODELIST")
+num_nodes=${SLURM_NNODES}
+# devices_per_node=32
+MASTER_ADDR=$(echo "$nodes" | head -n 1)
+MASTER_PORT=41008
+JAX_COORDINATOR_PORT=41001
+export NEURON_RT_ROOT_COMM_ID="${MASTER_ADDR}:${MASTER_PORT}"
+# export NEURON_PJRT_PROCESSES_NUM_DEVICES=$(printf '%s,' $(seq 1 $num_nodes | xargs -I {} echo $devices_per_node) | sed 's/,$//')
+export NEURON_PJRT_PROCESS_INDEX=${SLURM_NODEID}
+export LD_LIBRARY_PATH="/opt/amazon/efa/lib/"
+export FI_LOG_LEVEL="warn"
+export FI_EFA_USE_DEVICE_RDMA="1"
+export FI_PROVIDER="efa"
+export FI_EFA_FORK_SAFE=1
+export LD_LIBRARY_PATH="$HOME/EFA/aws-ofi-nccl/lib:LD_LIBRARY_PATH"
+
+ARTIFACTS_PATH="/fsx/thiamha/fs/runs/artifacts"
+TEST_ARTIFACTS_PATH="${ARTIFACTS_PATH}/$CONDA_ENV_NAME/t_${TESTNAME}/${JOB_ID}"
+mkdir -p "$TEST_ARTIFACTS_PATH"
+
+echo "NEURON RUN DIRECTORY ${TEST_ARTIFACTS_PATH} ${SLURM_JOB_ID} ${RESUME_JOB_ID}"
+
+NEURON_DUMP_PATH=${TEST_ARTIFACTS_PATH}/neuron_dump
+HLO_DUMP_PATH=${TEST_ARTIFACTS_PATH}/hlo_dump
+
+export XLA_PYTHON_CLIENT_PREALLOCATE=false
+export XLA_PYTHON_CLIENT_ALLOCATOR=platform
+
+# export JAX_PLATFORMS=cpu
+
+#Perf Tuning Guideline here : https://github.com/NVIDIA/JAX-Toolbox/blob/main/rosetta/docs/PGLE.md
```

Comment:
why are GPU flags in TRN runs?

Reply:
This is just the checkpointing script, I did not do any cleanup here.